### PR TITLE
feat: improve service topology controller by using client.Client instead of kubernetes.Interface

### LINKED ISSUE #1559: [feature request]improve service topology controller by using client.Client instead of kubernetes.Interface
**What would you like to be added**:
In service topology controller, `kubernetes.Interface` is used for updating annotations of [endpoints](https://github.com/openyurtio/openyurt/blob/master/pkg/controller/servicetopology/adapter/endpoints_adapter.go#L49), [v1.endpointslices](https://github.com/openyurtio/openyurt/blob/master/pkg/controller/servicetopology/adapter/endpointslicev1_adapter.go#L61), [v1beta1.endpointslices](https://github.com/openyurtio/openyurt/blob/master/pkg/controller/servicetopology/adapter/endpointslicev1beta1_adapter.go#L61).

and it is recommended to use `client.Client` instead of `kubernetes.Interface` to update K8s object in `yurt-manager` controller because controller-runtime is imported by yurt-manager.

**Why is this needed**:
it is recommended to use `client.Client` instead of `kubernetes.Interface` to update K8s object in `yurt-manager` controller because controller-runtime is imported by yurt-manager.

**others**
/kind feature

### PR DESCRIPTION
**Pull request by zhy76**
feat: improve service topology controller by using client.Client instead of kubernetes.Interface

/kind feature
> /kind question
> /kind design
> /sig ai
> /sig iot
> /sig network
> /sig storage

#### What this PR does / why we need it:
In service topology controller, kubernetes.Interface is used for updating annotations of [endpoints](https://github.com/openyurtio/openyurt/blob/master/pkg/controller/servicetopology/adapter/endpoints_adapter.go#L49), [v1.endpointslices](https://github.com/openyurtio/openyurt/blob/master/pkg/controller/servicetopology/adapter/endpointslicev1_adapter.go#L61), [v1beta1.endpointslices](https://github.com/openyurtio/openyurt/blob/master/pkg/controller/servicetopology/adapter/endpointslicev1beta1_adapter.go#L61).

and it is recommended to use client.Client instead of kubernetes.Interface to update K8s object in yurt-manager controller because controller-runtime is imported by yurt-manager.
#### Which issue(s) this PR fixes:

Fixes #1559 

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/controller/servicetopology/endpoints/endpoints_controller.go [hunk 1]
```diff
@@ -82,12 +81,12 @@ func (r *ReconcileServicetopologyEndpoints) InjectClient(c client.Client) error
 }
```
Comment (rambohe-ch) [inline pkg/controller/servicetopology/endpoints/endpoints_controller.go:81]: we can use this `client.Client` to new `endpointsAdapter` as following:
```
	r.Client = c
        r.endpointsAdapter = adapter.NewEndpointsAdapter(c)
	return nil
```

then we can remove `InjectConfig(cfg *rest.Config)` function.

Comment (rambohe-ch) [reply pkg/controller/servicetopology/endpoints/endpoints_controller.go:81]: by the way, it seems that `client.Client` field is unused in `ReconcileServicetopologyEndpoints`, so we can remove it.

File: pkg/controller/servicetopology/endpointslice/endpointslice_controller.go [hunk 2]
```diff
@@ -85,18 +84,17 @@ var _ reconcile.Reconciler = &ReconcileServiceTopologyEndpointSlice{}
 // ReconcileServiceTopologyEndpointSlice reconciles a Example object
 type ReconcileServiceTopologyEndpointSlice struct {
 	client.Client
-	kubeClient               kubernetes.Interface
 	endpointsliceAdapter     adapter.Adapter
 	isSupportEndpointslicev1 bool
 }
 
 func (r *ReconcileServiceTopologyEndpointSlice) InjectConfig(cfg *rest.Config) error {
```
Comment (rambohe-ch) [inline pkg/controller/servicetopology/endpointslice/endpointslice_controller.go:91]: ditto

File: pkg/controller/servicetopology/endpoints/endpoints_controller.go (hunk 1 above)
Comment (zhy76) [reply pkg/controller/servicetopology/endpoints/endpoints_controller.go]: The client.Client is used here: https://github.com/openyurtio/openyurt/blob/c730e6445dd9fbeba9e106a1f0bd367a8fb0ea37/pkg/controller/servicetopology/endpoints/endpoints_controller.go#L126

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**